menpo/model/pca.py

```diff
@@ -977,3 +944,37 @@ def __str__(self):
             self.noise_variance(), self.noise_variance_ratio(),
             self.n_components, self.components.shape)
         return str_out
+
+
+def extract_data(samples, n_samples=None, verbose=False):
```

Comment:
@jalabort this is nice how you cleaned it up but in light of @patricksnape s PR #528 we should probably use his generalised solution. Not sure which order we should bring these in?